app/src/main/java/io/gnosis/safe/ui/transactions/TransactionListFragment.kt

```diff
@@ -59,6 +59,11 @@ class TransactionListFragment : SafeOverviewBaseFragment<FragmentTransactionList
 
                 binding.progress.isVisible = loadState.refresh is LoadState.Loading && adapter.itemCount == 0
                 binding.refresh.isRefreshing = loadState.refresh is LoadState.Loading && adapter.itemCount != 0
+
+                val refresh = loadState.refresh
+                if (refresh is LoadState.Error) {
+                    handleError(refresh.error)
+                }
```

Comment:
Actually this is a perfect use case for `let`. I think everyone uses it as a `null` checker but I think it really shines when you need an immutable reference to an object brought into scope. 

Not that there is anything wrong with doing this, the way you did. Just sharing a thought :D

Reply:
how would you use `let` here? ```loadState.refresh.let { if(it is LoadState.Error) handle(it) } ```?

Reply:
yeah, just like that you would only have to change `handleError(it.error)`

Reply:
it would be perfect to you use `takeIf{ it is LoadState.Error}?.let{...}` but it sucks that smart casts don't work in that scenario :/